lms/views/email.py

```diff
@@ -115,6 +123,13 @@ def set_preferences(self):
         email_preferences.sat = params.get("sat") == "on"
         email_preferences.sun = params.get("sun") == "on"
 
+        if email_preferences.mention_email_feature_enabled:
+            # Only set the mention_email_subscribed preference if the feature is enabled
+            # This avoids setting it to False if the feature is not yet enabled
+            email_preferences.mention_email_subscribed = (
+                params.get("mention_email_subscribed") == "on"
+            )
```

Comment:
Ok, and this is the branch triggered when submitting the form, which I see it saves all settings at once in a single submit.

In that case, I reckon we should save the original settings conditionally, only if the logged-in user is an instructor. Does it make sense?

Reply:
Yes, that's missing here, we should only save the digest ones for instructors :+1: